src/dmclock_server.h

```diff
@@ -783,15 +787,34 @@ namespace crimson {
       {
 	assert(_erase_age >= _idle_age);
 	assert(_check_time < _idle_age);
-	cleaning_job =
+	idle_erase_job =
 	  std::unique_ptr<RunEvery>(
 	    new RunEvery(check_time,
-			 std::bind(&PriorityQueueBase::do_clean, this)));
+			 std::bind(&PriorityQueueBase::do_idle_erase, this)));
       }
 
 
       ~PriorityQueueBase() {
 	finishing = true;
+
+	DataGuard g(data_mtx);
```

Comment:
locking in a destructor is generally unnecessary, unless it's to synchronize with the shutdown of an internal thread. you don't need to provide mutual exclusion between the destructor and public functions, because it would already be undefined behavior for one thread to call functions on an object that's being destructed in another thread